File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/Events/SeatTypeEvent.cs
using System;
using ENode.Eventing;

namespace ConferenceManagement
{
    public abstract class SeatTypeEvent : DomainEvent<Guid>
    {
        public Guid SeatTypeId { get; private set; }
        public SeatTypeInfo SeatTypeInfo { get; private set; }

        public SeatTypeEvent() { }
        public SeatTypeEvent(Guid seatTypeId, SeatTypeInfo seatTypeInfo)
        {
            SeatTypeId = seatTypeId;
            SeatTypeInfo = seatTypeInfo;
        }
    }
}

File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/events/SeatTypeEvent.java
package com.microsoft.conference.management.domain.events;

import com.microsoft.conference.management.domain.models.SeatTypeInfo;
import lombok.Getter;
import lombok.Setter;
import org.enodeframework.eventing.DomainEvent;

@Getter
@Setter
public abstract class SeatTypeEvent extends DomainEvent<String> {
    private String seatTypeId;
    private SeatTypeInfo seatTypeInfo;

    public SeatTypeEvent() {
    }

    public SeatTypeEvent(String seatTypeId, SeatTypeInfo seatTypeInfo) {
        this.seatTypeId = seatTypeId;
        this.seatTypeInfo = seatTypeInfo;
    }
}
